packages/core/src/components/tabs2/tabs.tsx

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright 2015 Palantir Technologies, Inc. All rights reserved.
+ * Licensed under the BSD-3 License as modified (the “License”); you may obtain a copy
+ * of the license at https://github.com/palantir/blueprint/blob/master/LICENSE
+ * and https://github.com/palantir/blueprint/blob/master/PATENTS
+ */
+
+import * as classNames from "classnames";
+import * as PureRender from "pure-render-decorator";
+import * as React from "react";
+// import { findDOMNode } from "react-dom";
+
+import { AbstractComponent } from "../../common/abstractComponent";
+import * as Classes from "../../common/classes";
+import * as Keys from "../../common/keys";
+import { IProps } from "../../common/props";
+import { safeInvoke } from "../../common/utils";
+
+import { ITab2Props, Tab2, TabId } from "./tab";
+import { generateTabPanelId, generateTabTitleId, TabTitle } from "./tabTitle";
+
+// <Tabs id="tabs">
+//     <Tab id="a" title="Alpha">
+//         <h1>first panel</h1>
+//     </Tab>
+//     <Tab id="b" title="Beta">
+//         <h1>second panel</h1>
+//     </Tab>
+//     <input type="text" placeholder="Search..." />
+// </Tabs>
+
+// TODO
+// vertical key bindings? up/dn
+
+type TabElement = React.ReactElement<ITab2Props & { children: React.ReactNode }>;
+
+const TAB_SELECTOR = `.${Classes.TAB}`;
+
+export interface ITabs2Props extends IProps {
+    /**
+     * Whether the selected tab indicator should animate its movement.
+     * @default true
+     */
+    animate?: boolean;
+
+    /**
+     * Initial selected tab `id`. Note that this prop refers only to `<Tab>` children;
+     * other types of elements are ignored.
+     * @default first tab
+     */
```

Comment:
FWIW based on the language here I didn't _immediately_ grok how this prop would play into the usage (feels obvious now, but I was initially reading this as some kind of identifier that was tossed to whichever element was currently selected, rather than the ID of the tab I should render as initially selected). Might be worthwhile to massage the messaging here just a little.